TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -883,14 +882,25 @@ public static void CheckLibrary([NotNull] ProcedureDivision procedureDivision)
                 //TCRFUN_ONLY_PARAGRAPH_AND_PUBLIC_FUNC_IN_LIBRARY
                 if (!(child is FunctionDeclaration || child is Declaratives))
                 {
-                        DiagnosticUtils.AddError(child.CodeElement == null
-                                ? (child is Sentence
+                    Node node = child.CodeElement == null
+                        ? (child is Sentence
                             ? (child.Children.FirstOrDefault(c => c.CodeElement != null) ?? procedureDivision)
-                                    : procedureDivision)
-                                : child,
-                            "Inside a library only function declaration or declaratives are allowed " + child.Name +
-                            " / " + child.ID);
+                            : procedureDivision)
+                        : child;
+                    if (firstParagraphChecked)
+                    {
+                        // this case corresponds to SECTION declarations or statements not inside a paragraph
+                        string nodeInfo = " " + node.Name + " / " + node.ID;
+                        if (nodeInfo == "  / ")  nodeInfo = "";
+                        DiagnosticUtils.AddError(node,
+                            "Inside a library only function declaration or declaratives are allowed" + nodeInfo);
```

Comment:
Can you change the message to:
"A program which contains public procedure cannot contains section or statement not under a paragraph."

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1488/commits/6c457786fae470a996fff3b9f4509cce5e20546a